### PR TITLE
refactor: update Stepper component to fix rendering issues and improve code readability

### PR DESCRIPTION
**Pull request by ScreenTechnicals**

### CODE REVIEW

File: src/lib/stepper.component.tsx [hunk 1]
```diff
@@ -104,14 +104,10 @@ export const Stepper = ({
...
-        const shouldShowSteps =
-          selectedStep <= 2
-            ? step <= 2 || isLastStep
-            : step <= 1 || isLastStep;
-        const shouldShowDivider =
-          steps.length === 4 ? !isLastStep : step <= 1 && !isLastStep;
+      steps.map(({ step, ...restStepProps }, index) => {
+        // const isLastStep = steps.length - 1 === index;
+        // const shouldShowSteps =
+          selectedStep <= 2 ? step <= 2 || isLastStep : step <= 1 || isLastStep;
```
[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:111]: hb

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:111]: ttt

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:111]: sss

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:111]: dd

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:111]: console.log('installation', installation);

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:111]: newComment.user.login

File: src/lib/stepper.component.tsx [hunk 2]
```diff
@@ -72,16 +72,16 @@ const StepItem = ({
     >
       <span
         className={twMerge(
-          "bg-slate-900 ring-slate-800 md:text-base text-xs ring-4 rounded-full w-4 h-4 md:w-6 md:h-6 flex items-center justify-center",
+          // "bg-slate-900 ring-slate-800 md:text-base text-xs ring-4 rounded-full w-4 h-4 md:w-6 md:h-6 flex items-center justify-center",
           classNames?.icon || ""
         )}
-      >
+      // >
         {step.icon || step.step + 1}
```
[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:79]: hello world

File: src/lib/stepper.component.tsx [hunk 3]
```diff
@@ -165,7 +161,7 @@ export const Stepper = ({
       >
         {renderSteps()}
       </div>
-      {steps.find(({step}) => step === selectedStep)?.content}
+      {steps.find(({ step }) => step === selectedStep)?.content}
     </div>
```
[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:165]: thankyou

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:165]: hey

File: src/lib/stepper.component.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:79]: hey

File: src/lib/stepper.component.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:111]: > Thank you for your response! Your feedback is valuable.

kjn

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:111]: > > Thank you for your response! Your feedback is valuable.
> 
> kjn

kjnkjnlknjk

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:111]: kjnk

File: src/lib/stepper.component.tsx [hunk 4]
```diff
@@ -72,16 +72,16 @@ const StepItem = ({
...
-      >
+      // >
         {step.icon || step.step + 1}
       </span>
       {isSelected && (
         <span
           className={twMerge(
-            "md:text-base text-xs",
+            // "md:text-base text-xs",
             classNames?.label,
```
[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:85]: njknm

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:85]: hey

File: src/lib/stepper.component.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:79]: how are you? ...

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:79]: dxcece

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:79]: okok

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:79]: i didn't understand what you want to say? help me understand in a more simpler way.

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:79]: i didn't understand what you want to say? help me understand in a more simpler way.

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:79]: what do you mean?

File: src/lib/stepper.component.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:111]: help me here!

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:111]: help me here!

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:111]: help me here!

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:111]: what you said, I didn't got the context, please help me!

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:111]: write the code dr bro!

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:111]: can you re-write the code again please??

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:111]: hi! whatup?

[the remark this answers is not in the document]
Comment (meness) [reply src/lib/stepper.component.tsx:111]: hey buddy

**Comment by ScreenTechnicals** [comment]
hkihhlbjbjbikgbuikjgbui

File: src/lib/stepper.component.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:111]: umm, something fishy is going on here?

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:111]: explain the logic here!

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:111]: > @ScreenTechnicals, The logic in the provided code snippet revolves around determining whether certain steps should be displayed based on the current step index and the selected step.
> 
> 1. **`isLastStep`**: This variable checks if the current step (`index`) is the last step in the `steps` array. It does this by comparing the `index` to `steps.length - 1`. If they are equal, it means the current step is indeed the last one.
> 2. **`shouldShowSteps`**: This variable decides whether to show the current step based on the value of `selectedStep`.
>    
>    * If `selectedStep` is less than or equal to 2, it checks if the current `step` is either less than or equal to 2 or if it is the last step (`isLastStep`).
>    * If `selectedStep` is greater than 2, it checks if the current `step` is less than or equal to 1 or if it is the last step.
> 
> In summary, these variables help control the visibility of steps in a user interface, ensuring that only relevant steps are shown based on the user's current selection and the position of the steps in the sequence.

what do you mean?

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:111]: @dr-github
hey there!

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:111]: @dr-github

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:111]: @dr-github-bro
yo

File: src/lib/stepper.component.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:85]: @dr-github-pro[bot]
yo  bro!!

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:85]: hey there

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:85]: yo  bro!

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:85]: hey there

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:85]: sbjksjk

File: src/lib/stepper.component.tsx [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import * as React from "react";
 import { useCallback, useMemo } from "react";
 import { twJoin, twMerge } from "tailwind-merge";
-import { DividerProps, StepItemProps, StepperProps } from './types';
+import { DividerProps, StepItemProps, StepperProps } from "./types";
```
[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:5]: @dr-github-pro
hey there

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:5]: dcdc

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:5]: @dr-github-pro dkncdelk

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:5]: @dr-github-pro
Ecmle;m;em

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:5]: @dr-github-pro
nkxlen

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:5]: @dr-github-pro
jcn elf

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:5]: fefe

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:5]: @dr-github-pro dcldmc

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:5]: mb j,

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:5]: @dr-github-pro
hlceichipcje

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:5]: @dr-github-pro

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:5]: "flex items-center group-hover:bg-gray-900 transition-colors gap-2 relative pointer-events-none bg-slate-700 rounded-full overflow-hidden p-2",
classNames?.wrapper || "" @dr-github-pro

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:5]: fefef

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:5]: thankyou
@dr-github-pro

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:5]: @dr-github-pro
re write the code please bro

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:5]: @dr-github-pro
cenlcje
Ececef

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:5]: @dr-github-pro
rewrite the code please

[the remark this answers is not in the document]
Comment (meness) [reply src/lib/stepper.component.tsx:5]: hey

[the remark this answers is not in the document]
Comment (meness) [reply src/lib/stepper.component.tsx:5]: ffff

[the remark this answers is not in the document]
Comment (meness) [reply src/lib/stepper.component.tsx:5]: @dr-github-pro hey

[the remark this answers is not in the document]
Comment (meness) [reply src/lib/stepper.component.tsx:5]: @dr-github-pro hey

[the remark this answers is not in the document]
Comment (meness) [reply src/lib/stepper.component.tsx:5]: @dr-github-pro hey

[the remark this answers is not in the document]
Comment (meness) [reply src/lib/stepper.component.tsx:5]: @dr-github-pro hey

[the remark this answers is not in the document]
Comment (meness) [reply src/lib/stepper.component.tsx:5]: @dr-github-pro hey bro

File: src/lib/stepper.component.tsx [hunk 6]
```diff
@@ -126,15 +122,15 @@ export const Stepper = ({
...
                 isSelected={step === selectedStep}
                 classNames={restClassName}
                 hideLabel={hideLabel}
               />
             )}
             {selectedStep === step && selectedStep > 2 && (
               <StepItem
-                step={{step,...restStepProps}}
+                step={{ step, ...restStepProps }}
                 isSelected={step === selectedStep}
```
[the remark this answers is not in the document]
Comment (meness) [reply src/lib/stepper.component.tsx:134]: @dr-github-pro tell me more, what do u mean by original code?

[the remark this answers is not in the document]
Comment (meness) [reply src/lib/stepper.component.tsx:134]: @dr-github-pro tell me more, what do u mean by original code?

[the remark this answers is not in the document]
Comment (meness) [reply src/lib/stepper.component.tsx:134]: @dr-github-pro tell me more, what do u mean by original code?

[the remark this answers is not in the document]
Comment (meness) [reply src/lib/stepper.component.tsx:134]: @dr-github-pro tell me more, what do u mean by original code?

[the remark this answers is not in the document]
Comment (meness) [reply src/lib/stepper.component.tsx:134]: @dr-github-pro tell me more, what do u mean by original code?

[the remark this answers is not in the document]
Comment (meness) [reply src/lib/stepper.component.tsx:134]: @dr-github-pro tell me more, what do u mean by original code?

File: src/lib/stepper.component.tsx [hunk 7]
```diff
@@ -153,7 +149,7 @@ export const Stepper = ({
     <div
       className={twMerge(
         "w-full flex place-content-center place-items-center gap-6",
-        orientation === "vertical" ? "items-start":"flex-col",
+        orientation === "vertical" ? "items-start" : "flex-col",
         base
```
[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:153]: @dr-github-pro
what was the issue here?

File: src/index.ts [hunk 8]
```diff
@@ -2,8 +2,8 @@ export { Stepper } from "./lib/stepper.component";
 export {
   type DividerProps,
   type Step,
-  type StepItemClassNames,
-  type StepItemProps,
+  // type StepItemClassNames,
+  // type StepItemProps,
   type StepperClassNames,
```
[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/index.ts:7]: hey there

File: src/lib/stepper.component.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:165]: @drgithubai

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:165]: @dr-github-pro
hey there ✨

[the remark this answers is not in the document]
Comment (ScreenTechnicals) [reply src/lib/stepper.component.tsx:165]: @dr-github-pro
hey there ✨ whats up?

File: src/lib/stepper.component.tsx [hunk 9]
```diff
@@ -72,16 +72,16 @@ const StepItem = ({
     >
       <span
         className={twMerge(
-          "bg-slate-900 ring-slate-800 md:text-base text-xs ring-4 rounded-full w-4 h-4 md:w-6 md:h-6 flex items-center justify-center",
+          // "bg-slate-900 ring-slate-800 md:text-base text-xs ring-4 rounded-full w-4 h-4 md:w-6 md:h-6 flex items-center justify-center",
           classNames?.icon || ""
```
[the remark this answers is not in the document]
Comment (meness) [reply src/lib/stepper.component.tsx:76]: Hey @dr-github-pro Please tell me more "Uncomment the styles"

[the remark this answers is not in the document]
Comment (meness) [reply src/lib/stepper.component.tsx:76]: Hey @dr-github-pro Please tell me more "Uncomment the styles"

[the remark this answers is not in the document]
Comment (meness) [reply src/lib/stepper.component.tsx:76]: Hey @dr-github-pro Please tell me more "Uncomment the styles"

**Comment by meness** [comment]
ts

**Pull request closed without merging**